cms/taxonomy/management/commands/sync_topics.py

```diff
@@ -0,0 +1,203 @@
+    topics_response.raise_for_status()
+
+    return topics_response.json().get("items", [])
+
+
+def _extract_subtopic_links(raw_topics: list[dict[str, Any]]) -> list[tuple[str, str]]:
+    """Return a list of tuples of any subtopic links and their parent topic IDs found in raw_topics."""
+    return [
+        (subtopic_link, raw_topic["id"])
+        for raw_topic in raw_topics
+        # NOTE: This assumes we can simply call the links in the response as they are provided,
+        # perhaps switch to building the URL ourselves
+        if (
+            (subtopic_link := raw_topic.get("links", {}).get("subtopics", {}).get("href"))
+            and raw_topic.get("subtopics_ids")
+        )
+    ]
+
+
+def _sync_with_fetched_topics(fetched_topics: list[dict[str, str]]):
+    """For each fetched topic, decide if it needs to be created, updated, or left as is."""
+    _check_for_duplicate_topics(fetched_topics)
+
+    updated_count = 0
+    created_count = 0
+    for fetched_topic in fetched_topics:
+        if existing_topic := _get_topic(fetched_topic["id"]):
+            if not _topic_matches(existing_topic, fetched_topic):
+                _update_topic(existing_topic, fetched_topic)
+                updated_count += 1
+        else:
+            _create_topic(fetched_topic)
+            created_count += 1
+
+    logger.info("Saved %d new topic(s)", created_count)
+    logger.info("Updated %d existing topic(s)", updated_count)
+
+
+def _check_for_duplicate_topics(fetched_topics: list[dict[str, str]]) -> None:
+    topic_ids = set()
+    for topic in fetched_topics:
+        if topic["id"] in topic_ids:
+            raise RuntimeError(f"Received duplicate topic ID in API responses, topic IDs must be unique: {topic['id']}")
+        topic_ids.add(topic["id"])
+
+
+def _get_topic(topic_id: str) -> Topic | None:
+    """Fetches a Topic record by its primary key (id), or returns None if none is found."""
+    return Topic.objects.filter(id=topic_id).first()
+
+
+def _topic_matches(existing_topic: Topic, fetched_topic: dict[str, str]) -> bool:
+    """Compares: title, description, removed status, Parent ID.
+    If all these match, the function returns True; otherwise False.
+    """
+    existing_parent = existing_topic.get_parent()
+    existing_parent_id = existing_parent.id if existing_parent else None
+
+    return (
+        fetched_topic["title"] == existing_topic.title
+        and fetched_topic.get("description") == existing_topic.description
+        and not existing_topic.removed
+        and existing_parent_id == fetched_topic["parent_id"]
+    )
+
+
+def _update_topic(existing_topic: Topic, fetched_topic: dict[str, str]) -> None:
+    """Sets: title, description, removed = False (since the topic is present in the external data,
+    it cannot be considered removed). Parent/child changes: If the new parent_id is different from the old one,
+    we do a tree move of the topic to the new parent.
```

Comment:
I suggest something like:
```
"""
Updates an existing topic with data from an external source.

  - Updates `title`, `description`, and sets `removed = False` (as it is present in external data).
  - If the `parent_id` differs from the current parent, moves the topic to the new parent.
  - Logs changes and warnings when moving topics.
"""

Reply:
Addressed in https://github.com/ONSdigital/dis-wagtail/pull/104/commits/89fb9c9c8c63153bba489e9abedc3084ec285310